File: Server-code/internal/models/notification.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// Notification 系统通知：任务指派/完成/反馈/盯办等事件推送给相关人员
type Notification struct {
	ID          uuid.UUID  `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	RecipientID uuid.UUID  `gorm:"type:uuid;index" json:"recipient_id"`
	SenderID    *uuid.UUID `gorm:"type:uuid" json:"sender_id"`
	Sender      *User      `gorm:"foreignKey:SenderID" json:"sender,omitempty"`
	NoteID      *uuid.UUID `gorm:"type:uuid;index" json:"note_id"`
	IssueID     *uuid.UUID `gorm:"type:uuid;index" json:"issue_id"`    // 需求26：issue 评论通知关联的问题
	Type        string     `gorm:"type:varchar(30);index" json:"type"` // task_assigned/task_completed/task_feedback/task_remind/issue_comment/system
	Title       string     `gorm:"type:varchar(200)" json:"title"`
	Content     string     `gorm:"type:text" json:"content"`
	IsRead      bool       `gorm:"default:false;index" json:"is_read"`
	IsDeleted   bool       `gorm:"default:false" json:"is_deleted"`
	ReadAt      *time.Time `json:"read_at"`
	CreatedAt   time.Time  `json:"created_at"`
}

func (Notification) TableName() string {
	return "notifications"
}

// ChatMessage 用户之间的一对一聊天消息
type ChatMessage struct {
	ID         uuid.UUID  `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	SenderID   uuid.UUID  `gorm:"type:uuid;index" json:"sender_id"`
	ReceiverID uuid.UUID  `gorm:"type:uuid;index" json:"receiver_id"`
	NoteID     *uuid.UUID `gorm:"type:uuid" json:"note_id"`
	Type       string     `gorm:"type:varchar(20);default:'text';index" json:"type"` // text 文本 / image 图片 / file 文件
	Content    string     `gorm:"type:text" json:"content"`
	FileName   string     `gorm:"type:varchar(255)" json:"file_name"`
	FilePath   string     `gorm:"type:varchar(500)" json:"file_path"`
	FileSize   int64      `gorm:"default:0" json:"file_size"`
	MimeType   string     `gorm:"type:varchar(100)" json:"mime_type"`
	IsRead     bool       `gorm:"default:false;index" json:"is_read"`
	ReadAt     *time.Time `json:"read_at"`
	CreatedAt  time.Time  `json:"created_at"`
}

func (ChatMessage) TableName() string {
	return "chat_messages"
}

File: Web-Front/src/types/notification.ts
export interface NotificationItem {
  id: string
  recipient_id: string
  sender_id?: string
  sender?: {
    id: string
    username: string
    real_name?: string
    avatar?: string
  }
  note_id?: string
  /** 需求26：issue 评论通知关联的问题 id */
  issue_id?: string
  type: 'task_assigned' | 'task_completed' | 'task_feedback' | 'task_remind' | 'issue_comment' | 'issue_new' | 'system'
  title: string
  content: string
  is_read: boolean
  is_deleted: boolean
  read_at?: string
  created_at: string
}

export interface ChatMessageItem {
  id: string
  sender_id: string
  receiver_id: string
  note_id?: string
  type: 'text' | 'image' | 'file'
  content: string
  file_name?: string
  file_path?: string
  file_size?: number
  mime_type?: string
  is_read: boolean
  read_at?: string
  created_at: string
}

export interface ConversationItem {
  peer_id: string
  peer_name?: string
  peer_avatar?: string
  last_msg: string
  last_type?: string
  last_at: string
  unread: number
}

export interface ReminderItem {
  id: string
  note_id: string
  reminder_id: string
  target_id: string
  message: string
  remind_type: string
  is_acknowledged: boolean
  created_at: string
}

// ---------------- 群聊 ----------------

export interface GroupConversationItem {
  id: string
  name: string
  owner_id: string
  member_count: number
  unread: number
  last_msg: string
  last_type?: string
  last_at?: string | null
  created_at: string
}

export interface GroupMessageItem {
  id: string
  group_id: string
  sender_id: string
  /** 实时推送时附带的发送者姓名（历史消息通过 sender 关联查询） */
  sender_name?: string
  sender?: { id: string; username: string; name?: string; avatar?: string }
  type: 'text' | 'image' | 'file'
  content: string
  /** 被 @ 的成员 ID 列表（后端过滤为有效群成员；无 @ 时为空/缺省） */
  mentions?: string[]
  file_name?: string
  file_path?: string
  file_size?: number
  mime_type?: string
  created_at: string
}

export interface GroupMemberItem {
  id: string
  group_id: string
  user_id: string
  role: 'owner' | 'member'
  user?: {
    id: string
    username: string
    name?: string
    avatar?: string
    is_active?: boolean
    department?: { id: string; name: string }
  }
  joined_at: string
  last_read_at?: string | null
}

// 群文件夹文件（成员上传/下载，上传者或群主可删除）
export interface GroupFileItem {
  id: string
  group_id: string
  uploader_id: string
  uploader?: {
    id: string
    username: string
    name?: string
    avatar?: string
    department?: { id: string; name: string }
  }
  /** 实时推送时附带的上传者姓名（历史列表通过 uploader 关联查询） */
  uploader_name?: string
  file_name: string
  file_path: string
  file_size: number
  mime_type: string
  download_count: number
  created_at: string
}
